### PR TITLE
Serial-2

### PR DESCRIPTION
**Pull request by tjjfvi**

### CODE REVIEW

File: examples/Logo.ts [hunk 1]
```diff
@@ -24,6 +24,7 @@ export default renderFunction(
     const main = escad.cube({ size: outerSize })
     const inner = escad.cube({ size: innerSize })
     const corner = escad.cube({ size: outerSize, shift: 1 })
+    // @ts-ignore
```
Comment (vjjft) [inline examples/Logo.ts]: Did this just become necessary due to changes, and if so, why?

Can this be addressed without `@ts-ignore`?

File: packages/messages/src/serializeConnection.ts [hunk 2]
```diff
@@ -1,7 +1,43 @@
 
-import { Connection, transformConnection } from "."
-import { filterConnection } from "./filterConnection"
-import flatted from "flatted"
+import { createEventEmitter } from "."
+import { $unknown } from "../../serial/dist"
```
Comment (vjjft) [inline packages/messages/src/serializeConnection.ts]: fix import

File: packages/serial/src/$unknown.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  Object.getPrototypeOf(async function*(){}),
+])
+
+export const $unknown = new Serializer<unknown>({
+  *s(value){
+    const type = getType(value)
+    const $value = registeredTypes[type]
+    if(!$value)
+      throw new Error(`Cannot serialize value of type ${type}`)
+    yield* $string.s(type)
```
Comment (vjjft) [inline packages/serial/src/$unknown.ts:19]: consider using single byte for standard types (e.g. `string`,`number`,`null`,...)

File: packages/serial/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+registerType("object", $object)
+registerType("string", $string)
+registerType("number", $number)
+registerType("boolean", $boolean)
+registerType("null", $constant(null))
+registerType("undefined", $constant(undefined))
+registerType("Uint8Array", $buffer)
+registerType("Array", $array($unknown))
+registerType("Map", $map($unknown, $unknown))
+registerType("Set", $set($unknown))
```
Comment (vjjft) [inline packages/serial/src/index.ts:23]: extract to separate file

File: packages/core/src/Component.ts [hunk 5]
```diff
@@ -15,21 +15,23 @@ export interface Component<I extends any[], O extends Thing> {
   readonly hierarchy?: HierarchyProp,
   readonly overrideHierarchy?: boolean,
   readonly showOutput?: boolean,
+  readonly info: Record<string, any>,
```
Comment (vjjft) [inline packages/core/src/Component.ts:18]: `info`?

File: packages/core/src/RealmComponent.ts [hunk 6]
```diff
@@ -23,6 +23,7 @@ export const RealmComponent = {
           func: component.func,
           overrideHierarchy: component.overrideHierarchy,
           hierarchy: component.hierarchy,
+          info: component.info,
```
Comment (vjjft) [inline packages/core/src/RealmComponent.ts:26]: `info`?

File: packages/messages/tests/__snapshots__/serializeConnection.test.ts.snap [hunk 7]
```diff
@@ -1,20 +1,3 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
-exports[` 1`] = `
```
Comment (vjjft) [inline packages/messages/tests/__snapshots__/serializeConnection.test.ts.snap:3]: ?

File: packages/serial/src/$unknown.ts (hunk 3 above)
Comment (tjjfvi) [reply packages/serial/src/$unknown.ts:19]: Deferring

**Pull request merged**